src/components/camera-tool.js

```diff
@@ -100,12 +106,38 @@ AFRAME.registerComponent("camera-tool", {
     this.el.removeEventListener("stateadded", this.stateAdded);
   },
 
+  remove() {
+    if (this.cameraSystem) {
+      this.cameraSystem.deregister(this.el);
+    }
+  },
+
   stateAdded(evt) {
     if (evt.detail === "activated") {
       this.takeSnapshotNextTick = true;
     }
   },
 
+  focus(el, track) {
+    this.lookAt(el);
+
+    if (track) {
+      this.trackTarget = el;
+
+      this.trackTarget.addEventListener("componentremoved", e => {
+        if (e.detail.name === "position" && e.target === this.trackTarget) {
+          this.trackTarget = null;
+        }
+      });
+    }
+  },
+
+  lookAt(el) {
+    const targetPos = new THREE.Vector3();
```

Comment:
You are creating a new `THREE.Vector3` every tick here, I would rewrite this as:
```
lookAt: (function() {
    const targetPos = new THREE.Vector3();
    return function(el) {
        this.el.object3D.lookAt(targetPos.setFromMatrixPosition(el.object3D.matrixWorld));
    }
})(),